src/components/Editor.vue

```diff
@@ -559,14 +559,14 @@ export default {
 			this.document = document
 
 			this.syncError = null
-			const editable = !this.readOnly
+			const editable = !this.readOnly && !this.hasConnectionIssue
 			if (this.$editor.isEditable !== editable) {
 				this.$editor.setEditable(editable)
 			}
 		},
 
 		onSync({ steps, document }) {
-			this.hasConnectionIssue = false
+			this.hasConnectionIssue = !this.$providers[0].wsconnected || this.$syncService.pushError > 0
```

Comment:
I think I'm missing something obvious, but I don't understand why we should evaluate `this.$syncService.pushError > 0` here. `onSync()` is called when `SyncService` emits `'sync'`, which in my understanding happen either if we pushed steps successfully (which means `pushError = 0`), or if we received steps from the server (which is not related to `pushError`).